File: plugin/src/test/java/de/infolektuell/gradle/jextract/GradleJextractPLuginTest.java
package de.infolektuell.gradle.jextract;

import de.infolektuell.gradle.jextract.extensions.JextractExtension;
import org.gradle.api.Project;
import org.gradle.testfixtures.ProjectBuilder;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertNotNull;

class GradleJextractPluginTest {
    @Test
    void registersExtension() {
        Project project = ProjectBuilder.builder().build();
        project.getPlugins().apply("application");
        project.getPlugins().apply(GradleJextractPlugin.PLUGIN_NAME);
        assertNotNull(project.getExtensions().findByType(JextractExtension.class));
    }
}

File: plugin/src/test/kotlin/de/infolektuell/gradle/jextract/GradleJextractPluginTest.kt
package de.infolektuell.gradle.jextract

import de.infolektuell.gradle.jextract.extensions.JextractExtension
import org.gradle.testfixtures.ProjectBuilder
import kotlin.test.Test
import kotlin.test.assertNotNull

/**
 * A simple unit test for the 'org.example.greeting' plugin.
 */
class GradleJextractPluginTest {
    @Test fun `plugin registers extension`() {
        // Create a test project and apply the plugin
        val project = ProjectBuilder.builder().build()
        project.plugins.apply(GradleJextractPlugin.PLUGIN_NAME)
        assertNotNull(project.extensions.findByType(JextractExtension::class.java))
    }
}
